platforms/sui/protocols/cctp/src/circleBridge.ts

```diff
@@ -0,0 +1,256 @@
+    } = suiCircleObjects(network as "Mainnet" | "Testnet");
+
+    if (!contracts.cctp?.tokenMessenger) 
+      throw new Error(
+        `Circle Token Messenger contract for domain ${chain} not found`,
+      );
+
+    if (!contracts.cctp?.messageTransmitter) 
+      throw new Error(
+        `Circle Message Transmitter contract for domain ${chain} not found`,
+      );
+
+    this.usdcId = usdcId;
+    this.usdcTreasuryId = usdcTreasury;
+    this.tokenMessengerId = contracts.cctp?.tokenMessenger;
+    this.messageTransmitterId = contracts.cctp?.messageTransmitter;
+    this.tokenMessengerStateId = tokenMessengerState;
+    this.messageTransmitterStateId = messageTransmitterState;
+  }
+
+  async *transfer(
+    sender: AccountAddress<C>,
+    recipient: ChainAddress,
+    amount: bigint,
+  ): AsyncGenerator<SuiUnsignedTransaction<N, C>> {
+    const tx = new TransactionBlock();
+
+    const destinationDomain = circle.circleChainId.get(
+      this.network,
+      recipient.chain,
+    )!;
+
+    const [usdcStruct] = await SuiPlatform.getCoins(this.provider, sender, this.usdcId);
+
+    if (!usdcStruct) {
+      throw new Error('No USDC in wallet');
+    }
+
+    // It thinks balance is not a property when it is
+    // Maybe need to update SDK?
+    /* @ts-ignore */
+    if (usdcStruct.balance < amount) {
+      throw new Error('Amount exceeds USDC balance');
+    }
+
+    const [coin] = tx.splitCoins(
+      usdcStruct.coinObjectId,
+      [amount]
+    );
+
+    tx.moveCall({
+      target: `${this.tokenMessengerId}::deposit_for_burn::deposit_for_burn`,
+      arguments: [
+        coin!,
+        tx.pure.u32(destinationDomain), // destination_domain
+        tx.pure.address(recipient.address.toString()), // mint_recipient
+        tx.object(this.tokenMessengerStateId), // token_messenger_minter state
+        tx.object(this.messageTransmitterStateId), // message_transmitter state
+        tx.object("0x403"), // deny_list id, fixed address
+        tx.object(this.usdcTreasuryId) // treasury object Treasury<USDC>
+      ],
+      typeArguments: [this.usdcId],
+    });
+
+    yield this.createUnsignedTx(tx, "Sui.CircleBridge.Transfer")
+  }
+
+  async isTransferCompleted(message: CircleBridge.Message): Promise<boolean> {
+    const tx = new TransactionBlock();
+
+    tx.moveCall({
+      target: `${this.messageTransmitterId}::state::is_nonce_used`,
+      arguments: [
+        tx.object(this.messageTransmitterStateId),
+        tx.pure.u32(message.sourceDomain),
+        tx.pure.u64(message.nonce),
+      ],
+    });
+
+    const result = await this.provider.devInspectTransactionBlock({
+      sender: "0x0000000000000000000000000000000000000000000000000000000000000000",
+      transactionBlock: tx,
+    });
+
+    try {
+      /* @ts-ignore */
+      const isNonceUsed = Boolean(result.results![0].returnValues![0][0][0]);
+      return isNonceUsed;
+    } catch (e) {
+      console.error(`Error reading if nonce was used: ${e}`);
```

Comment:
Should we let this exception bubble up instead of logging in the SDK? This seems like a "shouldn't ever happen" type of exception, so maybe it's okay to just keep the `console.log` here. Just wanted to get your thoughts